tap_partnerize/client.py

```diff
@@ -6,28 +6,94 @@
 from typing import Any, Callable, Iterable
 
 import requests
+import datetime
+import csv
+import logging
+import io
 from singer_sdk.authenticators import BasicAuthenticator
-from singer_sdk.helpers.jsonpath import extract_jsonpath
 from singer_sdk.streams import RESTStream
+from singer_sdk.pagination import BaseAPIPaginator
+from requests import Response
+from singer_sdk.plugin_base import PluginBase as TapBaseClass
+from singer_sdk._singerlib import Schema
 
 _Auth = Callable[[requests.PreparedRequest], requests.PreparedRequest]
 SCHEMAS_DIR = Path(__file__).parent / Path("./schemas")
 
 
+class DayChunkPaginator(BaseAPIPaginator):
+    """A paginator that increments days in a date range."""
+
+    def __init__(self, start_date, increment=1, *args: Any, **kwargs: Any) -> None:
+        super().__init__(start_date)
+        self._value = datetime.datetime.strptime(start_date, "%Y-%m-%d")
+        self._end = datetime.datetime.today()
+        self._increment = increment
+
+    @property
+    def end_date(self):
+        """Get the end pagination value.
+
+        Returns:
+            End date.
+        """
+        return self._end
+
+    @property
+    def increment(self):
+        """Get the paginator increment.
+
+        Returns:
+            Increment.
+        """
+        return self._increment
+
+    def get_next(self, response: Response):
+        return self.current_value + datetime.timedelta(days=self.increment) if self.has_more(response) else None
+
+    def has_more(self, response: Response) -> bool:
+        """Checks if there are more days to process.
+
+        Args:
+            response: API response object.
+
+        Returns:
+            Boolean flag used to indicate if the endpoint has more pages.
+        """
+        return self.current_value < self.end_date
+
+
+def set_none_or_cast(value, expected_type):
+    if value == '' or value is None:
+        return None
+    elif not isinstance(value, expected_type):
+        return expected_type(value)
+    else:
+        return value
+
+
 class PartnerizeStream(RESTStream):
     """Partnerize stream class."""
 
-    # TODO: Set the API's base URL here:
-    url_base = "https://api.mysample.com"
+    url_base = "https://api.partnerize.com"
 
-    # OR use a dynamic url_base:
-    # @property
-    # def url_base(self) -> str:
-    #     """Return the API URL root, configurable via tap settings."""
-    #     return self.config["api_url"]
+    def __init__(
```

Comment:
I thought about that, but the only reason I couldn't figure it out was because I need to set `next_page_token` to `self.config.get("start_date", "")` and I didn't have access to `self` outside of the `__init__()` method. Unless there's another way to get access to the config that I'm not aware of?

Reply:
i think you can do that with a `@property` decorator then, analogous to [this](https://github.com/voxmedia/tap-instagram/blob/master/tap_instagram/client.py#L20-L23) or even some stuff you have in your code already

Reply:
Good call. I made that change now. I wasn't sure if the property would return the original config value for the `start_date` every time it was called, rather than incrementing. But I guess the property function just sets that initial value.